LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t03_lib.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - In the augmenting constructor's body, an `augmented()` call executes the
+///     augmented constructor's body in the same parameter scope that the
+///     augmenting body is executing in. The expression has type `void` and
+///     evaluates to `null`.
+///
+/// @description Checks that in the augmenting constructor's body, an
+/// `augmented()` call executes the augmented constructor's body in the same
+/// parameter scope that the augmenting body is executing in.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A10_t03.dart';
+
+augment class C {
+  augment C(int x, [int y = 0]) {
+    _log += "C($x, $y);";
```

Comment:
The really weird thing is that if we do `y++` just before `augmented()` then the augmented constructor body should have a `y` whose value is the same as the value of the parameter `y` in this body. So `y++` here should determine the outcome of an evaluation of `y` in the augmented constructor's body, not just the outcome of an evaluation of `y` in this body.

Reply:
`x++` added

Reply:
Thanks!